returnn/tf/layers/basic.py

```diff
@@ -2391,17 +2391,32 @@ def __init__(self, axes, padding, value=0, mode="constant", **kwargs):
     :param str|list[str] axes: e.g. "F" etc. see :func:`Dataset.get_axes_from_description`.
     :param list[(int,int)]|(int,int)|int padding: how much to pad left/right in each axis
     :param int|float value: what constant value to pad, with mode=="constant"
-    :param str mode: "constant", "reflect" or "symmetric"
+    :param str mode: "constant", "reflect", "symmetric" and "replication"
     """
-    from returnn.tf.util.data import DimensionTag
     super(PadLayer, self).__init__(**kwargs)
     axes = self.input_data.get_axes_from_description(axes)
     padding = self._transform_padding(padding=padding, axes=axes)
     paddings = [(0, 0)] * len(range(self.input_data.batch_ndim))
     for i, a in enumerate(axes):
       paddings[a] = padding[i]
     mode = mode.upper()
-    self.output.placeholder = tf.pad(self.input_data.placeholder, paddings=paddings, mode=mode, constant_values=value)
+    if mode == "REPLICATION":
+      assert len(padding) == 1, "Not implemented otherwise yet"
+      assert len(axes) == 1, "Not implemented otherwise yet"
+      pad_left_shape = [1] * self.input_data.batch_ndim
+      pad_left_shape[axes[0]] = padding[0][0]
+      pad_left = tf.gather(self.input_data.placeholder, 0, axis=axes[0])
+      pad_left = tf.expand_dims(pad_left, axis=axes[0])
+      pad_left = tf.tile(pad_left, pad_left_shape)
+      pad_right_shape = [1] * self.input_data.batch_ndim
+      pad_right_shape[axes[0]] = padding[0][1]
+      pad_right = tf.gather(
+        self.input_data.placeholder, tf.shape(self.input_data.placeholder)[axes[0]] - 1, axis=axes[0])
+      pad_right = tf.expand_dims(pad_right, axis=axes[0])
+      pad_right = tf.tile(pad_right, pad_right_shape)
+      self.output.placeholder = tf.concat([pad_left, self.input_data.placeholder, pad_right], axis=axes[0])
```

Comment:
Can you implement that in `returnn/tf/util.py` or so as a function `pad_replicate` or so?